vision/module/location.py

```diff
@@ -257,32 +424,41 @@ def _circle_detection(self) -> np.ndarray:
         -------
         ndarray - circles detected in image.
         """
-        # Size of the blur kernel
-        BLUR_SIZE = 5
+        BLUR_SIZE = 5  # Size of the blur kernel
+        # NOTE: May need increase w/ higher resolution, only tested on 480p data set
+        BRIGHTNESS_OFFSET = 42  # offset for brightness magnitude calculation
+        DEPTH_THRESH = 5000  # (in mm), tolerated dist. to filter circles
+
+        ## Image Pre-processing ##
+        # Increase brightness of image
+        bright = self._increase_brightness(offset=BRIGHTNESS_OFFSET)
 
         # Grayscale
-        gray = cv2.cvtColor(src=self.img, code=cv2.COLOR_RGB2GRAY)
+        bright_gray = cv2.cvtColor(src=bright, code=cv2.COLOR_RGB2GRAY)
 
         # Guassian Blur / Median Blur
-        # blur = cv2.GaussianBlur(src=gray, ksize=(BLUR_SIZE, BLUR_SIZE), sigmaX=0)
-        blur = cv2.medianBlur(gray, 15)
+        blur = cv2.GaussianBlur(src=bright_gray, ksize=(BLUR_SIZE, BLUR_SIZE), sigmaX=0)
+        # blur = cv2.medianBlur(gray, 15)
+        # blur = cv2.bilateralFilter(src=gray, d=11, sigmaColor=75, sigmaSpace=75)
 
         # Laplacian Transform / ksize = 3 for Guassian / ksize = 1 for Median
-        laplacian = cv2.Laplacian(src=blur, ddepth=cv2.CV_8U, ksize=1)
+        laplacian = cv2.Laplacian(src=blur, ddepth=cv2.CV_8U, ksize=3)
         laplacian = np.uint8(laplacian)
+        self.laplacianimg = np.copy(laplacian)  # HACK: REMOVE THIS
 
-        # Hough Circle Detection
-        self.circles = cv2.HoughCircles(
+        ## Hough Circle Detection ##
+        self.circles = cv2.HoughCircles(  # NOTE: Params may need partial rework, only tested on 480p data
             image=laplacian,
             method=cv2.HOUGH_GRADIENT,
             dp=1,
-            minDist=8,
-            param1=75,
-            param2=24,
+            minDist=4,
```

Comment:
May need to increase minimum distance, otherwise this could result in a ton of circles on higher res images.